tests/rustdoc-gui/src/test_docs/lib.rs

```diff
@@ -156,7 +156,7 @@ pub enum AnEnum {
     WithVariants { and: usize, sub: usize, variants: usize },
 }
 
-#[doc(keyword = "CookieMonster")]
+#[doc(keyword = "auto")]
```

Comment:
Why this change?

Reply:
The test now fails if the value isn't a valid keyword.

Reply:
Well, that brings this question: why this change?

Reply:
Previously the "is it a valid keyword?" test was done by the lint. It has now moved to `check_doc_keyword` and it runs in several test cases where the lint did not.

Reply:
No I mean, `doc(keyword)` was voluntarily left "open" in what kind of keyword we could define. You changed this behaviour. That's the part I don't understand. I don't mind changing it away from being a lint.

Reply:
The lint checked this:
```rust
fn is_doc_keyword(s: Symbol) -> bool {
    s <= kw::Union
}
```
That hasn't changed (well, `sym::SelfTy` is now also accepted, for the exception). What has changed is how often this check is being applied.

Reply:
I see, my bad.